### ISSUE TITLE
[App/Android] The landing predictor wipes its pinned prediction and cached winds on every reconnect, then flies the rest of the descent with zero wind

### ISSUE THREAD

**Issue by tinkerbug** [Tinkerbug-Robotics/TinkerRocket#1056]
**From the 2026-09-02 iOS ↔ Android app review** — landing-prediction lifecycle across a BLE reconnect. Part of #1035.

**Refs:** `TinkerRocketAndroid/core/session/src/main/kotlin/com/tinkerbug/tinkerrocket/session/LandingPredictor.kt:66`, `TinkerRocketAndroid/app/src/main/kotlin/com/tinkerbug/tinkerrocket/app/OfflineMapsScreen.kt:78`, `TinkerRocketApp/TinkerRocketApp/Models/LandingPredictor.swift:110`

### Claim
Android's `LandingPredictor.attach()` opens by calling its own `detach()`, which nulls the pinned prediction, the cached wind profile and `lastWindFetchAtMs`. iOS's `attach` only swaps the Combine subscription and clears nothing; on iOS only `detach()` resets state. The Android map re-attaches on every new `DeviceSession`, and the reconnect ladder constructs a new session object per reconnect, so a BLE drop and recovery wipes both. `prefetchWindIfNeeded` then refuses to fetch outside READY/PRELAUNCH/UNKNOWN, so the wind is never re-fetched in flight.

### Evidence
`LandingPredictor.kt:66-67` `public fun attach(session: DeviceSession, profileStore: RocketProfileStore) { detach()`, and `:81-85` `_prediction.value = null; _windProfile.value = null; _windFetchError.value = null; landed = false; lastWindFetchAtMs = null`. The class doc at `:26-28` states the intent this defeats: "if LoRa drops mid-flight the last prediction stays pinned so the recovery walk has a target". iOS `LandingPredictor.swift:111-114` is `// Drop any prior subscription before re-attaching.` / `cancellables.removeAll()` / `self.device = device` and nothing else.

Re-attach trigger: `OfflineMapsScreen.kt:78-82`, `val predictor = remember { ... LandingPredictor(container.fleetScope) }` with `DisposableEffect(session) { session?.let { predictor.attach(...) }; onDispose { predictor.detach() } }`. `FleetManager.kt:431` `adopt` calls `sessionFactory.create(` at `:451`, which is `AppContainer.kt:219` `val session = DeviceSession(` — a new object every reconnect.

Wind gate: `LandingPredictor.kt:182-183` `val preflight = t.state == "READY" || t.state == "PRELAUNCH" || t.state == "UNKNOWN"; if (!preflight) return`. In flight the wire state is `INFLIGHT` (`BsLogCsvGenerator.kt:71`), so no refetch happens. With no profile the cast substitutes a single zero-speed layer, `DriftCast.kt:345-348` `val windToUse = wind ?: WindProfile(layers = listOf(WindLayer(0.0, 0.0, 0.0))`, and the radius falls back to `ASSUMED_WIND_WHEN_UNKNOWN_MPS` = 2.0 m/s × descent seconds (`:320`, `:375-376`).

### Consequence
The lasting case is the backgrounded mid-flight drop from #829. Wind is fetched on the pad with the Map tab open, the rocket flies, the operator's screen locks during descent, and the link drops and recovers while the activity is stopped. The Recomposer is paused, so MapTab is never unwound (`MainActivity.kt:139` `activeDevice != null ->` reads true before and after), and on resume `DisposableEffect(session)` re-keys onto the new session and wipes both the pinned prediction and the pad wind. Every remaining descent cast is then a straight drop to the ground track directly beneath the rocket. On a 45 s main descent in 7 m/s of wind that discards roughly 300 m of drift, while the uncertainty ring only grows to about 95 m, so the ring signals degradation but does not cover the miss. Bounded honestly: a rocket-side LoRa blackout never rebuilds the BLE session, `lastValidRocketFix` still rehydrates from the fleet, live predictions resume on reconnect, and a drop while still on the pad self-heals because READY re-triggers the fetch. In a foreground drop the pin is lost for a different reason — `FleetManager.kt:477-481` removes the device and nulls `activeDeviceId`, so MapTab and its remembered predictor are destroyed outright.

### Fix
Mirror iOS in `LandingPredictor.kt:66`: `attach` should cancel and re-launch the telemetry collector and leave `_prediction`, `_windProfile` and `lastWindFetchAtMs` alone, with the clearing split out of `detach()` into an explicit `reset()` used only when the operator selects a different rocket. Change `OfflineMapsScreen.kt:79-82`'s `onDispose` to the unsubscribe-only path so a session swap does not clear either. That alone does not cover the foreground drop, so also hoist the predictor out of the `remember` at `OfflineMapsScreen.kt:78` into `AppContainer`, keyed by rocket identity, so MapTab's teardown no longer takes the pinned prediction with it. Keeping the cached profile is what makes this safe to do without touching `prefetchWindIfNeeded`'s gate — no mid-descent fetch is added, so the blocking `HttpURLConnection` on the fleet thread stays off the flight path.

**Verification:** Two independent refuters challenged this finding and neither refuted it; both voted to keep the severity at medium. They narrowed it in two ways. First, the mechanism is broader than the `DisposableEffect` key: `MainActivity.kt:139` gates the whole connected UI on `activeDevice != null`, so a foreground drop discards the remembered predictor regardless, and fixing `attach` alone would not restore the pin there. Second, impact is bounded because the recovery anchor and live predictions both survive, and the windless cast inflates the displayed uncertainty rather than hiding that it degraded.
